src/main/java/org/reaktivity/nukleus/mqtt/internal/stream/MqttServerFactory.java

```diff
@@ -616,128 +997,147 @@ private void onMqttPublish(
                 final MessageConsumer newTarget = router.supplyReceiver(newInitialId);
                 if (publishStream != null)
                 {
-                    publishStream.doMqttDataEx(decodeTraceId, payload, dataEx);
+                    publishStream.doMqttDataEx(decodeTraceId, authorization, payload, dataEx);
                     correlations.put(newReplyId, publishStream);    // TODO: do we need to clean up correlations onAbort()?
                 }
                 else
                 {
-                    final MqttPublishStream serverStream = new MqttPublishStream(newTarget,
-                        newRouteId, newInitialId, newReplyId, 0);
+                    final MqttPublishStream newPublishStream = new MqttPublishStream(newTarget,
+                            newRouteId, newInitialId, newReplyId, 0);
 
-                    serverStream.doBegin(decodeTraceId, affinity);
+                    newPublishStream.doApplicationBegin(decodeTraceId, authorization, affinity);
 
-                    serverStream.doMqttDataEx(decodeTraceId, payload, dataEx);
+                    newPublishStream.doMqttDataEx(decodeTraceId, authorization, payload, dataEx);
 
-                    correlations.put(newReplyId, serverStream);
-                    publishers.put(topicName, serverStream);
+                    correlations.put(newReplyId, newPublishStream);
+                    publishers.put(topicName, newPublishStream);
                 }
             }
         }
 
-        private void onMqttDisconnect(
+        private void onDecodeDisconnect(
+            long traceId,
             MqttDisconnectFW disconnect)
         {
             /* process reason code */
+            doNetworkEnd(traceId, authorization);
         }
 
-        private void doBegin(
+        private void onDecodeError(
             long traceId,
-            long affinity)
+            long authorization,
+            int reasonCode,
+            boolean isConnectError)
         {
-            final BeginFW begin = beginRW.wrap(writeBuffer, 0, writeBuffer.capacity())
-                .routeId(routeId)
-                .streamId(replyId)
-                .traceId(traceId)
-                .affinity(affinity)
-                .build();
+            cleanupStreams();
+            if (isConnectError)
```

Comment:
Can we use `connected` field here instead and remove `isConnectError ` parameter?